src/HDF5.jl

```diff
@@ -157,28 +156,28 @@ function Base.setindex!(parent::Union{File,Group}, val, path::Union{AbstractStri
     write(parent, path, val; pv...)
 end
 
-
-
 # end of high-level interface
 
-
 include("api_midlevel.jl")
 
-
 #API.h5s_get_simple_extent_ndims(space_id::API.hid_t) = API.h5s_get_simple_extent_ndims(space_id, C_NULL, C_NULL)
 
-
 # Functions that require special handling
 
 const libversion = API.h5_get_libversion()
 
 ### Property manipulation ###
-get_access_properties(d::Dataset)   = DatasetAccessProperties(API.h5d_get_access_plist(d))
-get_access_properties(f::File)      = FileAccessProperties(API.h5f_get_access_plist(f))
-get_create_properties(d::Dataset)   = DatasetCreateProperties(API.h5d_get_create_plist(d))
-get_create_properties(g::Group)     = GroupCreateProperties(API.h5g_get_create_plist(g))
-get_create_properties(f::File)      = FileCreateProperties(API.h5f_get_create_plist(f))
-get_create_properties(a::Attribute) = AttributeCreateProperties(API.h5a_get_create_plist(a))
+function get_global_property(name::Symbol)
+    local_context = get(task_local_storage(), :hdf5_context, CONTEXT)
+    property = getfield(local_context, name)
+    isvalid(property) ? property : nothing
+end
+get_access_properties(d::Dataset)   = something(get_global_property(:dataset_access), DatasetAccessProperties(API.h5d_get_access_plist(d)))
+get_access_properties(f::File)      = something(get_global_property(:file_access), FileAccessProperties(API.h5f_get_access_plist(f)))
+get_create_properties(d::Dataset)   = something(get_global_property(:dataset_create), DatasetCreateProperties(API.h5d_get_create_plist(d)))
+get_create_properties(g::Group)     = something(get_global_property(:group_create), GroupCreateProperties(API.h5g_get_create_plist(g)))
+get_create_properties(f::File)      = something(get_global_property(:file_create), FileCreateProperties(API.h5f_get_create_plist(f)))
+get_create_properties(a::Attribute) = something(get_global_property(:attribute_create), AttributeCreateProperties(API.h5a_get_create_plist(a)))
```

Comment:
> I only want the global property if the object (Dataset, File, etc) is not valid.

Mmh no, this will make `track_order` fail since `API.h5f_get_create_plist(f)` returns a `fpcl` with incorrect `track_order`, although valid fpcl. A valid global context property should preempt object properties.

Reply:
No. I would just copy that line into gen/api_defs.jl, run `julia --project=. gen/gen_wrappers.jl`, and then build a high level interface to copy property lists. I will probably add a bunch of API defs in another pull request if you just would like to wait for that.

Reply:
> Mmh no, this will make `track_order` fail since `API.h5f_get_create_plist(f)` returns a `fpcl` with incorrect `track_order`, although valid opened `fpcl`. A valid global context property should preempt object properties.

Restore the old methods then and create a new family of methods. `get_access_from_context` or `get_context_create` or `get_context_access` perhaps?